its/src/test/java/com/sonar/it/csharp/EnsureAllTestsRunTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * SonarSource :: C# :: ITs :: Plugin
+ * Copyright (C) 2011-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonar.it.csharp;
+
+import io.github.classgraph.AnnotationClassRef;
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ScanResult;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class EnsureAllTestsRunTest {
+
+  private static final String PACKAGE_NAME = "com.sonar.it.csharp";
+
+  private static final String TEST_SUITE_CLASS = PACKAGE_NAME + ".Tests";
+
+  /**
+   * In order for the tests to run when executing "mvn verify", all test classes need to be mentioned in the
+   * "@SuiteClasses" annotation on {@link com.sonar.it.csharp.Tests}
+   */
+  @Test
+  public void testClassesAreListedInTestsSuite() {
```

Comment:
Is not possible to replace the @SuiteClasses with an auto-detection mechanism in com.sonar.it.csharp.Tests?

Reply:
it may be possible, but I don't find a lot of value in doing this versus the current approach.

having in mind also the effort. Because all the annotations in tests (`@Rule`, `@ClassRule`, `@Before`) work together with `@Suite`. And the question is: why spend a day to rewrite this?

Reply:
Currently a test is needed to ensure that we don't forget to mention the test in `@SuiteClasses`. This is, as far as I can tell, a work around since it's not done automatically and we forget to add the new tests.

In my opinion there would be a big advantage in having this work out of the box (e.g. less code to maintain). I was hopping that the framework allows this but I'm not familiar with it and that's why I was asking.

If the effort to change this is too big I can understand avoiding this change.